### PR TITLE
Replace tslint with eslint

### PR DESCRIPTION
**Pull request by sprucely**
There are still a lot of warnings, most of which will require declaring interfaces and function return types.

### CODE REVIEW

**Comment by srkirkland** [comment]
Looks pretty clean, I'll test it out

File: src/CaesHelp/ClientApp/util/email.ts [hunk 1]
```diff
@@ -6,6 +6,6 @@
 /* tslint:disable-next-line:max-line-length */
     // This one is really permissive, perhaps replace with one from PEAKS
     // per: https://www.regular-expressions.info/email.html
-    const regex = new RegExp("^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$", "i");
+    const regex = new RegExp("^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,}$", "i");
```
Comment (jSylvestre) [inline src/CaesHelp/ClientApp/util/email.ts:9]: Is this right? If it is, how did it work before?

Comment (jSylvestre) [reply src/CaesHelp/ClientApp/util/email.ts:9]: NM, old one allows j@t\com which is bad...

**Comment by srkirkland** [comment]
I think we'll need to tweak some of the rules since they seem pretty harsh (no explicit any?), but since we aren't failing builds based on linting yet then I think we can keep these best practice rules for now and then update on an individual basis instead of trying to recreate our old rules.  I do like this setup though and the code changes you've made seem to work well.

**Pull request merged**